test/blackbox-tests/test-cases/variants/run.t

```diff
@@ -1,16 +1,62 @@
 Variant feature is auto enabled when virtual_modules is used
 
   $ dune build --root variants-without-using
-  File "dune", line 3, characters 1-25:
-  3 |  (virtual_modules foobar))
-       ^^^^^^^^^^^^^^^^^^^^^^^^
-  Error: 'virtual_modules' is only available since version 0.1 of the experimental variants feature
-  [1]
+  Entering directory 'variants-without-using'
 
   $ dune build --root variants-using
-  File "dune-project", line 2, characters 42-45:
-  2 | 
-  Error: Version 0.1 of in_development_do_not_use_variants is not supported.
-  Supported versions:
-  - 0.0
+  Entering directory 'variants-using'
+
+virtual libraries may not implement their virtual modules
+
+  $ dune build --root invalid-virtual-lib
+  Entering directory 'invalid-virtual-lib'
+  File "dune", line 3, characters 18-21:
+  3 |  (virtual_modules foo bar))
+                        ^^^
+  Error: Module Foo has an implementation, it cannot be listed here
   [1]
+
+  $ cd module-fields && ocaml test.ml
+  impl: true. modules_without_implementation: true. virtual_modules: true
+  File "dune", line 3, characters 33-34:
+  3 |  (modules_without_implementation m)
+                                       ^
+  Error: Module M has an implementation, it cannot be listed here
+  -------------------------
+  impl: true. modules_without_implementation: true. virtual_modules: false
+  File "dune", line 3, characters 33-34:
+  3 |  (modules_without_implementation m))
+                                       ^
+  Error: Module M has an implementation, it cannot be listed here
+  -------------------------
+  impl: true. modules_without_implementation: false. virtual_modules: true
+  File "dune", line 3, characters 18-19:
+  3 |  (virtual_modules m))
+                        ^
+  Error: Module M has an implementation, it cannot be listed here
+  -------------------------
+  impl: true. modules_without_implementation: false. virtual_modules: false
+  -------------------------
+  impl: false. modules_without_implementation: true. virtual_modules: true
+  File "dune", line 4, characters 18-19:
+  4 |  (virtual_modules m))
+                        ^
+  Error: These modules appear in the virtual_libraries and modules_without_implementation fields: 
+  - m
```

Comment:
To be consistent with the other error messages, we should use the capitalized name here

Reply:
I think that we used lowercase names elsewhere as well. I think that capitalization is better here however.